A beginners guide on how to become a blockchain developer (#2571)
section-engineering-education/engineering-education: content/articles/a-beginners-guide-on-how-to-become-a-blockchain-developer/index.md

2. Ethereum blockchain

The Ethereum blockchain is a decentralized and open-source smart contracts platform. The smart contract functionality that enables Dapps to be built on it has made it the most actively used blockchain platform. It currently uses PoW as its consensus mechanism but aiming to replace it with [Proof-of-Stake (PoS)](https://en.wikipedia.org/wiki/Proof_of_stake) due to the high energy consumption of PoW. 

It is also the blockchain technology that powers the popular cryptocurrency, ether (ETH).

Learn more about it [here](https://ethereum.org/en/).

3. Cardano blockchain

It is a decentralized and open-source blockchain platform that utilizes Proof-of-Stake (PoS) as its consensus mechanism. It is the blockchain platform that powers the popular cryptocurrency known as Ada. 
Unlike the Ethereum blockchain, Cardano is developed through evidence-based methods and peer-reviewed research. This means that proposed methods have to undergo academic-type research before being implemented.

This blockchain technology is built on the Haskell programming language commonly used in the banking and financial industry. 

#### Note

> Unless you have a really good reason, then you should use the Ethereum blockchain as it has the biggest developer community for building blockchain-based applications. As a developer, you wouldn't want to get stuck and not have a community to be able to ask questions.

Reviewer: ```suggestion
Unless you really have to use a different blockchain for development, I recommend that you use the Ethereum blockchain. It has the biggest developer community for building blockchain-based applications. As a developer, you definitely wouldn't want to get stuck and not have a community for support.
```

2. [React.js](https://reactjs.org/)

This is an open-source javascript library that enables programmers to build beautiful user interfaces. You can use React.js to build those beautiful user interfaces on the front-end.

3. [Ethers.js](https://docs.ethers.io/v5/)

It is a collection of libraries that allows you to interact with the Ethereum blockchain and its ecosystem.

4. [Metamask](https://metamask.io/)

It is a browser extension that allows a user to access their Ethereum wallets on the Ethereum blockchain. Users can buy, store, and swap ether tokens using this library.

5. [Node.js](https://nodejs.org/en/)

It is a back-end JavaScript runtime environment that executes javascript code outside a web browser. Decentralized applications usually use node.js to interact with smart contracts and public nodes.

6. [Ganache](https://www.trufflesuite.com/ganache)

It is a library developers can use for Ethereum blockchain development. It enables developers to develop, deploy, and test decentralized applications in a safe and deterministic environment without interacting with the main Ethereum blockchain.

### Which programming languages for blockchain?

1. Solidity

> This is the programming language that is used for creating Ethereum smart contracts.

Reviewer: This is probably the language that will definitely catch the eye of a reader. Perhaps provide a link to the Solidity page or a useful resource.

2. Javascript

It is used primarily for creating user interfaces. 

3. Go and C++

These two languages are used by the blockchain core developers to building the back-ends of blockchain systems.

4. Haskell

This is the programming language used by core blockchain developers in building the Cardano blockchain.

#### Note

> Solidity is still currently the most popular programming language of choice for many developers. This is due to its large developer community. Unless you have a specific use case or you're a blockchain core developer, please stick to Solidity.

Reviewer: ```suggestion
Solidity is still currently the most popular programming language of choice for many developers. This is due to its large developer community. Unless you have a specific use case or you're a blockchain core developer, I suggest you stick with Solidity.
```

### Use cases for blockchain

1. Finance

#### Cryptocurrencies

Today, there exist over 5,000 cryptocurrencies. This is according to [CoinMarketCap](https://coinmarketcap.com/), a company that tracks cryptocurrencies' market caps, rankings, prices, exchange volumes, and currency conversion. However, the most popular cryptocurrencies in this space include [Bitcoin](https://bitcoin.org/en/), [Ethereum](https://ethereum.org/en/), [Cardano](https://cardano.org/) and [Polkadot](https://polkadot.network/).

People tend to think that cryptocurrencies are the only blockchain's use case. This is not the case. But, it is the biggest use case for blockchain. 

> As a blockchain developer, you can build and launch one of your cryptocurrencies. But there is a catch. With the high number of cryptocurrencies out there, it is advisable to build and launch a coin that tries to solve real-world problems. Otherwise, it'll just exist and not be adopted by anyone. 

Reviewer: ```suggestion
As a blockchain developer, you can build and launch a token. You may even go all the way and build your own cryptocurrency. But there is a catch. With the high number of cryptocurrencies and tokens out there, it is advisable to develop a token (or cryptocurrency if feasible) that tries to solve real-world problems. Otherwise, it'll be hard to gain traction or survive in the long term.
```

Reviewer: I've made a suggestion on this line. Perhaps add a link that distinguishes between a token and a coin. And a single line mentioning the challenges of creating a cryptocurrency such as the resources needed and distrust from potential users due to the reputation of ICOs.

A good example is Cardano with their [Cardano Africa](https://africa.cardano.org/) project. They aim to use Ada (name of Cardano's cryptocurrency), to try and solve some of the financial problems currently experience in less developed countries such as accessibility to loans. This is a luxury enjoyed by many people in developed worlds. 

As a blockchain developer, you need to think along those lines.

#### Decentralized Finance (DeFi)

Currently, the adoption of DeFi is trending in the blockchain world. DeFi aims to replace the existing traditional finance system. It essentially takes financial products and moves them onto the blockchain ecosystem. A few examples of these products include savings, loans, and trading.

Taking a look at the 1-year chart on [DeFi Pulse](https://defipulse.com/), you will notice how the adoption of DeFi has skyrocketed over the last few months.

> Currently, most of the DeFi runs on the Ethereum blockchain. Popular DeFi project in the market today include [Maker](https://makerdao.com/en/), [Aave](https://aave.com/), [Compound](https://compound.finance/), [Polygon](https://polygon.technology/) and [Uniswap](https://uniswap.org/). Maker, Aave, and Compound are used as lending platforms, Polygon as a payment platform, and Uniswap for decentralized exchange (DEX).

Reviewer: ```suggestion
Currently, most of the DeFi runs on the Ethereum blockchain. Popular DeFi projects in the market today include [Maker](https://makerdao.com/en/), [Aave](https://aave.com/), [Compound](https://compound.finance/), [Polygon](https://polygon.technology/) and [Uniswap](https://uniswap.org/). Maker, Aave, and Compound are used as lending platforms, Polygon as a payment platform, and Uniswap for decentralized exchange (DEX).
```

> As a blockchain developer, you can build platforms such as these.

Reviewer: ```suggestion
As a blockchain developer and a techie in general, you should be ambitious enough to build such platforms.
```

2. Supply Chain

> The supply chain system involves tracking the movement of goods that end up turning into end products. Big companies such as Walmart use it to increase supply chain transparency.

Reviewer: ```suggestion
The supply chain system involves tracking the movement of goods or services from a source to an end-user. Big companies such as Walmart use it to increase supply chain transparency.
```

As a blockchain developer, you can build digital identity systems that secure digital IDs, degrees, certificates, transcripts, examinations, and badges, etc. on the blockchain.

4. Healthcare

In healthcare, blockchain can be used for record-keeping & medical history tracking, patient & drug identity validation, and to offer direct healthcare service payments. This [article](https://www.section.io/engineering-education/blockchain-as-a-revolution-in-healthcare/) explains clearly how blockchain technology is used to enable these services in healthcare. 

These are a few use cases for blockchain technology. It can also be used to secure title deeds in real estate and to secure voting systems from tamper in democratic countries.

### Wrapping Up

> In this article, I have explained how you can become a blockchain developer and take advantage of this incredibly scarce and lucrative field. It is wrong to assume that just because you have no prior coding experience, that this is not for you. Blockchain development is a skill you master over time. Just follow this guide and continuously try and practice. With blockchain development, practice makes perfect.

Reviewer: ```suggestion
In this article, I have explained how you can become a blockchain developer and take advantage of this lucrative field. It is wrong to assume that since you may have no prior coding experience, this niche is not for you. As with any skill, blockchain development is mastered over time. This guide gives a starting point. However, it is your responsibility to continuously learn, try and practice. As they say, practice makes perfect.
```

Reviewer: Consider adding references to add value for the reader
